tensorflow/cc/framework/gradients.cc

```diff
@@ -175,8 +175,14 @@ Status SymbolicGradientBuilder::Initialize() {
         "Must specify a gradient input for each output.");
   }
   std::vector<bool> reachable_nodes = GetReachableNodes();
-  // TODO(theflofly) Check that inputs_ are reachable from
-  // outputs_ using reachable_nodes
+  for (const Output& input : inputs_) {
+    if (!reachable_nodes[input.node()->id()]) {
+      return errors::InvalidArgument(
+        "Cannot compute the partial derivative for node '"
+        + input.node()->name()
```

Comment:
nit: don't use + to concat strings, you can just pass in multiple arguments: return InvalidArgument("foo", "bar")

Reply:
Indeed, I made the edit.